### DISCUSSION TITLE
Graphql query to fetch commits from all time

### DISCUSSION THREAD

**Discussion by itsvaibhavmishra** [anuraghazra/github-readme-stats#3494] [Q&A] [answered]
What is the Graphql query to fetch commits for all the time including the start of the github account to current date?

**Comment by qwerty541** [comment, marked as answer]
Hey, @itsvaibhavmishra! We do not use GraphQL for this purpose. Currently we fetch total commits count using GitHub HTTP API. Here is the code: https://github.com/anuraghazra/github-readme-stats/blob/master/src/fetchers/stats-fetcher.js#L170-L187. As far as i know it was implemented this way because it's impossible to fetch total commits count using GraphQL, only count for selected year, at least it was like that a few years ago, i didn't checked if something changed.

**Reply by rickstaa** [reply]
Absolutely, let's make it more polished:

@qwerty541 accurately highlights that we refrain from utilizing GraphQL for the retrieval of total commit data. The rationale behind this choice is that if we use GraphQL, we need to paginate through all the years of a user's GitHub history, potentially [triggering the GraphQL rate limits](https://docs.github.com/en/graphql/overview/rate-limits-and-node-limits-for-the-graphql-api) on our public instance. I have submitted a formal feature request [here](https://github.com/orgs/community/discussions/35675). Your endorsement and support for this feature request could draw attention to it on GitHub, potentially paving the way for implementing a solution that enables fetching all commits using GraphQL.
